src/Build/Instance/ProjectInstance.cs

```diff
@@ -1893,6 +1919,12 @@ private void TranslateAllState(ITranslator translator)
             translator.Translate(ref _isImmutable);
             translator.Translate(ref _evaluationId);
 
+            translator.Translate(ref _imports);
+            Imports = _imports.AsReadOnly();
```

Comment:
The assignments to `Imports` and `ImportsIncludingDuplicates` should only be made when `translator.Mode == TranslationDirectory.ReadFromStream`. It would avoid creating unneeded new list wrapper instances for the import properties when the same ProjectInstance is serialized multiple times (which might screw up user code if they put the readonly lists in dictionaries and such).

Reply:
Nice catch

Reply:
You know what, I see instance of stuff like this:
```
            // ignore EvaluatedItemElements. Only used by public API users, not nodes
            // ignore itemsByEvaluatedInclude. Only used by public API users, not nodes
```

So since this is only for the public API, it seems that I don't need to translate this at all.